backend/src/routes/users.js

```diff
@@ -98,10 +101,36 @@ const login = async (req, res) => {
     }
 };
 
+const generateNewPin = async (req, res) => {
+    try {
+        const phoneNumber = req.body.phoneNumber;
+        const validationErrors = utils.users.validateUserPhone(phoneNumber);
+        if (validationErrors) {
+            res.status(422).send(validationErrors);
+        } else {
+            const user = await db.users.getUser(phoneNumber);
+            if (user) {
+                const pin = randtoken.generate(4, '0123456789');
+                user.pin = pin;
+                await db.users.updateUser(user);
+                console.log(`New Pin Generated: ${pin}`);
```

Comment:
Would be good to use "PIN" instead of "Pin" everywhere in comments and texts as that is the right format for PIN (Personal Identification Number).